### PR TITLE
Explain more about protection for encryption keys (API data encryption at rest)

### PR DESCRIPTION
**Pull request by sftim**
This PR is heavily inspired by https://github.com/kubernetes/website/pull/33285 but has some custom changes.

- Explain importance of protecting keys and other material that can be
  used to decrypt data in etcd
- Revise the explanation for a non-KMS setup example
- Explicitly link to https://k8s.io/docs/tasks/administer-cluster/kms-provider/

### CODE REVIEW

File: content/en/docs/tasks/administer-cluster/encrypt-data.md [hunk 1]
```diff
@@ -311,7 +311,72 @@ Using envelope encryption creates dependence on the key encryption key, which is
...
 values would need to compromise etcd **and** the third-party KMS provider.
 
-## Write an encryption configuration file
+### Protection for encryption keys
+
+You should take appropriate measures to protect the confidential information that allows decryption,
+whether that is a local encryption key, or an authentication token that allows the API server to
+call KMS.
+
+Even with a provider managing the use and lifecycle of the key encryption key, you are still responsible
```
Comment (Ritikaa96) [inline content/en/docs/tasks/administer-cluster/encrypt-data.md:320]:
```suggestion
Even with a provider managing the usage and lifecycle of the encryption key, you are still responsible
```

Comment (Ritikaa96) [reply content/en/docs/tasks/administer-cluster/encrypt-data.md:320]: small nits

**Review by Ritikaa96 — CHANGES_REQUESTED** [review]

File: content/en/docs/tasks/administer-cluster/encrypt-data.md (hunk 1 above)
Comment (sftim) [reply content/en/docs/tasks/administer-cluster/encrypt-data.md]: I reworded - how's this?

File: content/en/docs/tasks/administer-cluster/encrypt-data.md [hunk 2]
```diff
@@ -403,6 +461,9 @@ Your config file contains keys that can decrypt the contents in etcd, so you mus
 permissions on your control-plane nodes so only the user who runs the `kube-apiserver` can read it.
 {{< /caution >}}
 
+You now have encryption in place for **one** control plane host. A typical
+Kubernetes cluster has multiple control plane host, so there is more to do.
```
Comment (kbhawkey) [inline content/en/docs/tasks/administer-cluster/encrypt-data.md:465]: A typical ... has multiple control plane hosts,

File: content/en/docs/tasks/administer-cluster/encrypt-data.md [hunk 3]
```diff
@@ -311,7 +311,72 @@ Using envelope encryption creates dependence on the key encryption key, which is
...
+# Do not run this in a session where you have set a random number
+# generator seed.
+[Convert]::ToBase64String((1..32|%{[byte](Get-Random -Max 256)}))
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+
+{{< note >}}
+Keep the encryption key confidential, including whilst you generate it and
```
Comment (kbhawkey) [inline content/en/docs/tasks/administer-cluster/encrypt-data.md:375]: you can probably drop the clause,
including whilst you generate it ....

Comment (sftim) [reply content/en/docs/tasks/administer-cluster/encrypt-data.md:375]: I like this emphasis; without it, people might only think about the protection after the key is in place.

Comment (kbhawkey) [reply content/en/docs/tasks/administer-cluster/encrypt-data.md:375]: nit: whilst -> while

**Pull request merged**